x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/predicate/operator/arithmetic/NegTests.java

```diff
@@ -112,14 +137,47 @@ public void testEdgeCases() {
 
             return;
         }
+        if (testCaseType == EsqlDataTypes.DATE_PERIOD) {
+            Period minPeriod = Period.of(Integer.MIN_VALUE, Integer.MIN_VALUE, Integer.MIN_VALUE);
+            assertNull(process(minPeriod));
+            assertCriticalWarnings(
+                "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
+                "java.lang.ArithmeticException: integer overflow"
+            );
+
+            Period maxPeriod = Period.of(Integer.MAX_VALUE, Integer.MAX_VALUE, Integer.MAX_VALUE);
+            Period negatedMaxPeriod = Period.of(-Integer.MAX_VALUE, -Integer.MAX_VALUE, -Integer.MAX_VALUE);
+            assertEquals(negatedMaxPeriod, process(maxPeriod));
+            return;
+        }
+        if (testCaseType == EsqlDataTypes.TIME_DURATION) {
+            Duration minDuration = Duration.ofSeconds(Long.MIN_VALUE, 0);
+            assertNull(process(minDuration));
+            assertCriticalWarnings(
+                "Line -1:-1: evaluation of [] failed, treating result as null. Only first 20 failures recorded.",
+                "java.lang.ArithmeticException: Exceeds capacity of Duration: 9223372036854775808000000000"
+            );
+
+            Duration maxDuration = Duration.ofSeconds(Long.MAX_VALUE, 0);
+            Duration negatedMaxDuration = Duration.ofSeconds(-Long.MAX_VALUE, 0);
+            assertEquals(negatedMaxDuration, process(maxDuration));
+
+            return;
+        }
         throw new AssertionError("Edge cases not tested for negation with type [" + testCaseType.typeName() + "]");
     }
 
-    private Object process(Number val) {
-        return toJavaObject(evaluator(new Neg(Source.EMPTY, field("val", typeOf(val)))).get().eval(row(List.of(val))), 0);
+    private Object process(Object val) {
+        if (testCase.allTypesAreRepresentable()) {
+            Neg neg = new Neg(Source.EMPTY, field("val", typeOf(val)));
+            return toJavaObject(evaluator(neg).get().eval(row(List.of(val))), 0);
+        } else {
+            Neg neg = new Neg(Source.EMPTY, new Literal(Source.EMPTY, val, typeOf(val)));
```

Comment:
This helper function would still need to be invoked from `process()`, since I'd like to avoid having the test cases "decide" how to "process" the negation. This being a test, I've just added a small comment on the else branch.